rackcity/utils/query_utils.py

```diff
@@ -0,0 +1,296 @@
+                    )
+                elif (
+                    filter_dict['max'] is not None
+                    and isinstance(filter_dict['max'], int)
+                    and (
+                        filter_dict['min'] is None
+                        or filter_dict['min'] == ""
+                    )
+                ):
+                    filter_args.append(
+                        {
+                            '{0}__lte'.format(filter_field): int(filter_dict['max'])  # noqa less than or equal to max
+                        }
+                    )
+                elif (
+                    int(filter_dict['max']) is not None
+                    and int(filter_dict['min']) is not None
+                ):
+                    range_value = (
+                        int(filter_dict['min']),
+                        int(filter_dict['max'])
+                    )
+                    filter_args.append(
+                        {
+                            '{0}__range'.format(filter_field): range_value  # noqa inclusive on both min, max
+                        }
+                    )
+                else:
+                    raise Exception(
+                        "Numeric filters must contain integer min, integer max, or both."
+                    )
+
+            elif filter_type == 'rack_range':
+                range_serializer = RackRangeSerializer(data=filter_dict)
+                if not range_serializer.is_valid():
+                    raise Exception(
+                        "Invalid rack_range filter: " +
+                        str(range_serializer.errors)
+                    )
+                filter_args.append(
+                    {
+                        'rack__rack_num__range':
+                        range_serializer.get_number_range()
+                    }
+                )
+                filter_args.append(
+                    {
+                        'rack__row_letter__range':
+                        range_serializer.get_row_range()  # noqa inclusive on both letter, number
+                    }
+                )
+
+            else:
+                raise Exception(
+                    "String field 'filter_type' must be either 'text', " +
+                    "'numeric', or 'rack_range'."
+                )
+
+    return filter_args
+
+
+def get_filtered_query(object_query, data):
+    try:
+        filter_args = get_filter_arguments(data)
+    except Exception as error:
+        return (
+            None,
+            JsonResponse(
+                {
+                    "failure_message":
+                    Status.ERROR.value + GenericFailure.FILTER.value,
+                    "errors": str(error)
+                },
+                status=HTTPStatus.BAD_REQUEST
+            )
+        )
+    for filter_arg in filter_args:
+        object_query = object_query.filter(**filter_arg)
+    return (object_query, None)
+
+
+def get_sorted_query(object_query, data):
```

Comment:
Refactored. Returns query set with sorts applied, and an optional 400 JsonResponse if something went wrong